src/api/db/data/20180109170324_re_encode_event_payloads.rb

```diff
@@ -0,0 +1,15 @@
+class ReEncodeEventPayloads < ActiveRecord::Migration[5.1]
+  def up
+    # Re-encode the events that creates notifications for using only one encoder (ActiveSupport::JSON)
+    Event::Base.notification_events.each do |event_type|
+      event_type.where(mails_sent: false).find_in_batches do |event|
+        event.set_payload(event.payload, event.payload_keys)
```

Comment:
Does this work? I thought that this data migration would have to first decode the payload using the old encoder, then encode it using the new encoder?

Reply:
What is encoded is JSON standard, and at the same time the parser is Yajl in both cases, I mean the 'old' state was that ActiveSupport::JSON uses Yajl for parsing if this is used and required in the gemfile. Is not the same case for the encoder part...

Reply:
I dont really understand.. In `Event::Base` you changed the encode/decode from:
`Yajl::Encoder.encode` and `Yajl::Encoder.decode` to `ActiveSupport::JSON.encode` and `ActiveSupport::JSON.decode` but if ActiveSupport::JSON calls Yajl then nothing has changed here right? (Apart from calling the encode/decode via an ActiveSupport wrapper). So why the need to re-encode?

Reply:
ActiveSupport does more things on top of what Yajl does, like escaping differently the html entities...

Reply:
@evanrolfe @mdeniz Personally I don't mind that much, but shouldn't this be a [data migration](https://github.com/openSUSE/open-build-service/tree/master/src/api/db/data)?

Reply:
It is a data migration? Because its in db/data, they also inherit from `ActiveRecord::Migration` just like normal migrations.

Reply:
It is a data migration... I used the data_migration generator 👍

Reply:
Argh, I am blind><
Ignore me